src/Integration.Vsix.UnitTests/Analysis/UserSettingsProviderTests.cs

```diff
@@ -130,46 +130,55 @@ public void FileChanges_EventsRaised()
         public void EnsureFileExists_CreatedIfMissing()
         {
             // Arrange
-            var fileMock = new Mock<IFile>();
-            var testSubject = new UserSettingsProvider(new TestLogger(), fileMock.Object, CreateMockFileMonitor("c:\\missingFile.txt").Object);
-            fileMock.Reset();
-            fileMock.Setup(x => x.Exists("c:\\missingFile.txt")).Returns(false);
+            var fileName = "c:\\missingFile.txt";
+
+            var fileSystemMock = new Mock<IFileSystem>();
+            fileSystemMock.Setup(x => x.File.Exists(fileName)).Returns(false);
+
+            var testSubject = new UserSettingsProvider(new TestLogger(), fileSystemMock.Object, CreateMockFileMonitor(fileName).Object);
+            fileSystemMock.Reset();
 
             // Act
             testSubject.EnsureFileExists();
             
             // Assert
-            fileMock.Verify(x => x.Exists("c:\\missingFile.txt"), Times.Once);
-            fileMock.Verify(x => x.WriteAllText("c:\\missingFile.txt", It.IsAny<string>()), Times.Once);
+            fileSystemMock.Verify(x => x.File.Exists(fileName), Times.Exactly(2));
```

Comment:
Interesting, looks like a known issue - https://github.com/moq/moq4/issues/733
There are two calls - one in UserSettingsProvider's ctor and one in EnsureFileExists method.
None of the Moq methods Reset() or Invocations.Clear() actually stop the Verify from thinking there were 2 executions.
It used to work in the past because we used to do the setup _after_ UserSettingsProvider's ctor, so there really was only 1 call. But we can't do that anymore because we'll get Null exception if it's not configured.
I'll remove the call to Reset since it's misleading.

Reply:
```
I've been expecting this issue to be raised eventually, thanks for reporting it. This is one of those not-so-clear-cut cases where it is perhaps up to personal interpretation what should happen.
``` 
:-)